### PR TITLE
Rev version

### PR DESCRIPTION
**Pull request by RickWinter**

### CODE REVIEW

**Review by ahsonkhan — APPROVED** [review]

**Review by antkmsft — APPROVED** [review]
I think that maybe we should undef AZ_SDK_VERSION_PRERELEASE, instead of making it empty.

**Review by antkmsft — COMMENTED** [review]
Because you can't write a condition `#ifdef AZ_SDK_VERSION_PRERELEASE == "anything"`, including `#ifdef AZ_SDK_VERSION_PRERELEASE == ""`.
But you can do `#ifdef AZ_SDK_VERSION_PRERELEASE` and then your code knows that you are compiling with Beta.
If you go with undefing it, which I strongly  suggest, update the comment above it.


Update: the comment is good, and thank you for undefing. I like the way you did it, with having both "" and then the undef.

**Comment by ahsonkhan** [comment]
Please hold off on merging this until we are ready to release.

File: sdk/inc/azure/core/az_version.h
Comment (antkmsft) [inline sdk/inc/azure/core/az_version.h:32]: For Doxygen:
```suggestion
#define AZ_SDK_VERSION_PRERELEASE
#undef AZ_SDK_VERSION_PRERELEASE
```

Comment (antkmsft) [reply sdk/inc/azure/core/az_version.h:32]: This is still not ideal. The docs will look as if it is defined. But if we go and undef it without ever defining, the info on AZ_SDK_VERSION_PRERELEASE will be missing from the docs.

Comment (ahsonkhan) [reply sdk/inc/azure/core/az_version.h:32]: Where do you see that?
I don't see `AZ_SDK_VERSION_PRERELEASE` anymore in the docs from the generated artifacts:
image

Comment (antkmsft) [reply sdk/inc/azure/core/az_version.h:32]: With this suggestion (define then undef next line), you will be able to:
image
image

Comment (ahsonkhan) [reply sdk/inc/azure/core/az_version.h:32]: It might be better that it is hidden and not visible in the docs, which we get by jus keeping undef. But your call Rick.

Comment (antkmsft) [reply sdk/inc/azure/core/az_version.h:32]: If we don't want in documentation, I suggest to wrap the whole thing, including the doxygen comment, as 
```
/*
/// Optional pre-release identifier. SDK is in a pre-release state when present.
#define AZ_SDK_VERSION_PRERELEASE "beta.1"
*/
``` 
- this way, note that it sets the format, and there is not Doxygen comment
If we have a doxygen comment, and then an undef entry but without a #define, the doxygen comment woud stickto the next documentable entity, if there's any in the future. But if we comment (note that we don't use the doxygen-style multi-line comment), then it is excluded alltogether.

Comment (ahsonkhan) [reply sdk/inc/azure/core/az_version.h:32]: I see. In that case, define/undef, like you suggested makes sense.

**Pull request merged**
